### PR TITLE
docs: initial content for Gestures

### PR DESCRIPTION
**Pull request by Ombuweb**

### CODE REVIEW

File: content/ui/gestures.md [hunk 1]
```diff
@@ -0,0 +1,419 @@
+---
+title: Gestures
+---
+<!-- TODO: SB for gestures -->
```
Comment (SeanKelly369) [inline content/ui/gestures.md:4]: Two TODOs

File: content/ui/gestures.md [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+| `getPointerCount()`| `number`| Gets the number of pointers in the event.| 
+| `getX()`| `number`| Gets the `x` coordinate of the event inside the view that triggered the event.| 
+| `getY()`| `number`| Gets the `y` coordinate of the event inside the view that triggered the event.| 
+
+<!-- /// flavor plain
+
+```ts
+import { GestureTypes, GestureEventData, Label } from '@nativescript/core'
+var label = new Label()
+label.on(GestureTypes.doubleTap, (args: GestureEventData) => {
```
Comment (SeanKelly369) [inline content/ui/gestures.md:86]: Is it worth pointing out that .on method can be used for all gesture types?

File: content/ui/gestures.md [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+
+<!-- /// flavor plain
+```xml
+  <Button  text="Swipe me" class="button"  swipe="{{ onSwipe }}"/>
+
+```
+ ```ts
+ import { SwipeGestureEventData } from '@nativescript/core'
+
+ onSwipe(args: SwipeGestureEventData) {
```
Comment (SeanKelly369) [inline content/ui/gestures.md:163]: Is it worthwhile pointing out the parameters in args that are useful for each gesture type?

File: content/ui/gestures.md [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+**Action**: A pan gesture occurs when a user presses down a component and immediately start moving it around. Pans are executed more slowly and allow for more precision and the screen stops responding as soon as the finger is lifted off it
+
+#### Pan gesture event data
+
+| Name | Type | Description |
+|------|------|-------------|
+| `deltaX` | `number` | Distance , in DIPs, moved in the `x` direction from previous position. |
+| `deltaY` | `number` | Distance , in DIPs, moved in the `x` direction from previous position.|
+| `state` | `GestureStateTypes` | Indicates the state of panning. See [GestureStateTypes](#gesturestatetypes) for available states. |
+<!-- TODO: CORRECT `state` values -->
```
Comment (SeanKelly369) [inline content/ui/gestures.md:210]: TODO

File: content/ui/gestures.md [hunk 5]
```diff
@@ -0,0 +1,419 @@
...
+
+#### Swipe gesture event data
+
+- `direction` : Direction of the swipe gesture.
+
+Available directions:
+
+- `right = 1`,
+- `left = 2`,
+- `up = 4`,
```
Comment (SeanKelly369) [inline content/ui/gestures.md:196]: Just checking, is 4 and 8 the correct values?

**Review by ashutosh887 — APPROVED** [review]
LGTM if we resolve the `TODO`'s

File: content/ui/gestures.md [hunk 6]
```diff
@@ -0,0 +1,419 @@
+---
+title: Gestures
+---
+<!-- TODO: SB for gestures -->
+<!-- TODO: Add flavor tabs for the gestures -->
+Gestures are the different kinds of ways a user interacts with a UI component. NativeScript supports the following gestures:
```
Comment (NathanWalker) [inline content/ui/gestures.md]:
```suggestion
Users can interact with UI components using gestures. NativeScript supports the following gestures:
```

File: content/ui/gestures.md [hunk 7]
```diff
@@ -0,0 +1,419 @@
...
+- [tap](#tap-gesture-in-nativescript)
+- [doubleTap](#double-tap-event-data)
+- [longPress](#long-press-gesture-in-nativescript)
+- [swipe](#swipe-gesture-in-nativescript)
+- [pan](#pan-gesture-in-nativescript)
+- [pinch](#pinch-gesture-in-nativescript)
+- [rotation](#rotate-gesture-in-nativescript)
+- [touch](#touch-gesture-in-nativescript)
+
+All the gesture events, except the `tap`, provide the following data in common:
```
Comment (NathanWalker) [inline content/ui/gestures.md]:
```suggestion
All gesture events, except `tap`, have the following data in common:
```

File: content/ui/gestures.md [hunk 8]
```diff
@@ -0,0 +1,419 @@
...
+
+## Tap gesture
+
+**Action**: Briefly touching a component.
+
+#### Tap gesture event data
+
+The `tap` gesture event only offers the `eventName` and `object` properties.
+
+The [TouchManager](/guide/ui-components/animations-in-nativescript#touchmanager-8-2) API offers cool animations that you can apply for the `tap` gesture.
```
Comment (NathanWalker) [inline content/ui/gestures.md]:
```suggestion
The [TouchManager](/guide/ui-components/animations-in-nativescript#touchmanager-8-2) offers a convenient and easy-to-use API including animation abilities that you can apply to the `tap` gesture (individually or even globally across your entire app for any wired tap gesture/event).
```

File: content/ui/gestures.md [hunk 9]
```diff
@@ -0,0 +1,419 @@
...
+
+**Action**: Briefly touching a component.
+
+#### Tap gesture event data
+
+The `tap` gesture event only offers the `eventName` and `object` properties.
+
+The [TouchManager](/guide/ui-components/animations-in-nativescript#touchmanager-8-2) offers a convenient and easy-to-use API including animation abilities that you can apply to the `tap` gesture (individually or even globally across your entire app for any wired tap gesture/event).
+
+To handle a `tap` event follow:
```
Comment (NathanWalker) [inline content/ui/gestures.md]:
```suggestion
Here are some example `tap` event bindings:
```

File: content/ui/gestures.md [hunk 10]
```diff
@@ -0,0 +1,419 @@
...
+```
+
+```html
+<label text="Tap here" (tap)="onTap($event)"></label>
+```
+/// -->
+
+## Double tap gesture
+
+**Action**: Two taps on a component in a quick succession.
```
Comment (NathanWalker) [inline content/ui/gestures.md]:
```suggestion
**Action**: Two taps on a component in quick succession.
```

File: content/ui/gestures.md [hunk 11]
```diff
@@ -0,0 +1,419 @@
...
+Available directions:
+
+- `right = 1`,
+- `left = 2`,
+- `up = 4`,
+- `down = 8`,
+
+## Pan gesture
+
+**Action**: A pan gesture occurs when a user presses down a component and immediately start moving it around. Pans are executed more slowly and allow for more precision and the screen stops responding as soon as the finger is lifted off it
```
Comment (NathanWalker) [inline content/ui/gestures.md]:
```suggestion
**Action**: A pan gesture occurs when a user presses down on a component and immediately starts moving it around. Pans are executed more slowly and allow for more precision. The event stops emitting as soon as the finger is lifted off it.
```

File: content/ui/gestures.md [hunk 12]
```diff
@@ -0,0 +1,419 @@
...
+```html
+<label text="Pan here" (pan)="onPan($event)"></label>
+```
+
+///
+ -->
+
+## Pinch gesture
+
+**Action**: A user touches a component with two fingers, then moves them towards each other or away from each other.
```
Comment (NathanWalker) [inline content/ui/gestures.md]:
```suggestion
**Action**: A user touches a component with two fingers, then moves them towards or away from each other.
```

File: content/ui/gestures.md [hunk 13]
```diff
@@ -0,0 +1,419 @@
...
+}
+```
+
+```html
+<label text="Pinch here" (pinch)="onPinch($event)"></label>
+```
+
+/// -->
+
+Possible usage: Zoom into content or out of content.
```
Comment (NathanWalker) [inline content/ui/gestures.md]:
```suggestion
Possible usage: Zoom in or out of content.
```

File: content/ui/gestures.md [hunk 14]
```diff
@@ -0,0 +1,419 @@
...
+<label text="Pinch here" (pinch)="onPinch($event)"></label>
+```
+
+/// -->
+
+Possible usage: Zoom in or out of content.
+
+## Rotate gesture
+
+**Action**: A user touches a component with two fingers, then rotate them simultaneously left or right.
```
Comment (NathanWalker) [inline content/ui/gestures.md]:
```suggestion
**Action**: A user touches a component with two fingers, then rotates them simultaneously left or right.
```

File: content/ui/gestures.md [hunk 15]
```diff
@@ -0,0 +1,419 @@
...
+
+You can handle multiple gestures as follows:
+
+```xml
+  <Button  text="TAP" class="button" longPress="{{ onLongPress }}" swipe="{{ onSwipe }}" tap="{{ onTap }}"  doubleTap="{{ onDoubleTap }}"/>
+```
+
+## Gesture Manipulations
+
+In some scenarios, you would want to disable the user interaction or to create more complex UI where some gestures are passing through the parents of the actual interactive zone. NativeScript provides some specific properties for handling similar cases as follows:
```
Comment (NathanWalker) [inline content/ui/gestures.md]:
```suggestion
In some scenarios, you may want to disable user interaction or create more complex UI interactions where some gestures are passing through the parents of the actual interactive zone. NativeScript provides some specific properties for handling similar cases as follows:
```

**Pull request merged**